projects/plugins/jetpack/src/class-deprecate.php

```diff
@@ -21,12 +22,34 @@ class Deprecate {
 	 */
 	private static $instance;
 
+	/**
+	 * An array of notices to display.
+	 *
+	 * @var array
+	 */
+	private $notices = array();
+
 	/**
 	 * Initialize the class.
 	 */
 	private function __construct() {
+		// phpcs:disable Squiz.PHP.CommentedOutCode.Found
+		// Example $this->notices content. Note that the key must end with '-deprecate-feature' to display in My Jetpack - see packages/my-jetpack/_inc/hooks/use-notification-watcher/use-deprecate-feature-notice.ts:
+		// array(
+		// 'my-admin-deprecate-feature' => array(
+		// 'title'     => __( "Retired feature: Jetpack's XYZ Feature", 'jetpack' ),
+		// 'message'   => __( "This feature is being retired and will be removed effective November, 2024. Please use the Classic Theme Helper plugin instead.", 'jetpack' ),
+		// 'link'      => array(
+		// 'label' => __( 'Learn more', 'jetpack' ),
+		// 'url'   => 'jetpack-support-xyz',
+		// ),
+		// 'condition' => 'show_feature_notice', // Method name to check if notice should show
```

Comment:
Maybe we could simplify things a bit by accepting that the condition will always be the same and relying on the cookie?
This way we could remove the `condition` requirement and instead introduce:
```
private function show_feature_notice( $id ) {
	return empty( $_COOKIE['jetpack_deprecate_dismissed'][$id] );
}
```

If anyone really wants to override this behaviour, they could add some special logic in `show_feature_notice`

Reply:
Yes, good idea, thank you. Cleaned up in 9aedbeed9380db4ac162d6a1313d5bda3d9a08e4